fire/SFEquationsMod.F90

```diff
@@ -0,0 +1,296 @@
+    real(r8), parameter :: a = 0.200395_r8
+    real(r8), parameter :: b = -0.8189_r8
+          
+    if (SAV < nearzero) then
+      OptimumPackingRatio = 0.0_r8 
+    else  
+      OptimumPackingRatio = a*(SAV**b)
+    end if
+          
+    end function OptimumPackingRatio
+    
+    !-------------------------------------------------------------------------------------
+  
+    real(r8) function MaximumReactionVelocity(SAV)
+      !
+      !  DESCRIPTION:
+      !  Calculates maximum reaction velocity in /min
+      !  Reaction velocity that would exist if fuel were free of moisture and 
+      !  contained mineral and alpha cellulose
+      !  From Equation 36 in Rothermel 1972; Fig. 12
+      !
+      
+      ! ARGUMENTS:
+      real(r8), intent(in) :: SAV ! fuel surface area to volume ratio [/cm]
+      
+      MaximumReactionVelocity = 1.0_r8/(0.0591_r8 + 2.926_r8*(SAV**(-1.5_r8)))
+    
+    end function MaximumReactionVelocity
+   
+    !-------------------------------------------------------------------------------------
+  
+    real(r8) function OptimumReactionVelocity(max_reaction_vel, SAV, beta_ratio)
+      !
+      !  DESCRIPTION:
+      !  Calculates optimum reaction velocity in /min
+      !  From Equation 38 in Rothermel 1972; Fig. 11
+      !
+      
+      ! ARGUMENTS:
+      real(r8), intent(in) :: max_reaction_vel ! maximum reaction velocity [/min]
+      real(r8), intent(in) :: SAV              ! fuel surface area to volume ratio [/cm]
+      real(r8), intent(in) :: beta_ratio       ! ratio of packing ratio to optimum packing ratio [0-1]
+      
+      ! LOCALS:
+      real (r8) :: a, a_beta ! intermediate variables
+      
+      ! Equations in Table A1 Thonicke et al. 2010
+      a = 8.9033_r8*(SAV**(-0.7913_r8))
+      a_beta = exp(a*(1.0_r8 - beta_ratio)) 
+      
+      OptimumReactionVelocity = max_reaction_vel*(beta_ratio**a)*a_beta
+      
+    end function OptimumReactionVelocity
+   
+    !-------------------------------------------------------------------------------------
+  
+    real(r8) function MoistureCoefficient(moisture, MEF)
+      !
+      !  DESCRIPTION:
+      !  Calculates the moisture dampening coefficient for reaction intensity
+      !  Based on Equation in table A1 Thonicke et al. 2010. 
+      !
+      
+      ! ARGUMENTS:
+      real(r8), intent(in) :: moisture ! fuel moisture [m3/m3]
+      real(r8), intent(in) :: MEF      ! fuel moisture of extinction [m3/m3]
+      
+      ! LOCALS:
+      real(r8) :: mw_weight ! relative fuel moisture/fuel moisture of extinction
+      
+      ! average values for litter pools (dead leaves, twigs, small and large branches), plus grass
+      mw_weight = moisture/MEF
+      
+      ! moist_damp is unitless
+      MoistureCoefficient = max(0.0_r8, (1.0_r8 - (2.59_r8*mw_weight) +                   &
+        (5.11_r8*(mw_weight**2.0_r8)) - (3.52_r8*(mw_weight**3.0_r8))))
+        
+      if (MoistureCoefficient > 1.0_r8) MoistureCoefficient = 1.0_r8
```

Comment:
Was this `if (MoistureCoefficient > 1.0_r8)` bit present in the original code? I'm not seeing it. It does make sense to have, though.

Also, could you use the same method instead of `max(0.0_r8, ...` above? It would be better for consistency and to make the above calculation easier to read.

Reply:
Oh no it wasn't - I saw this max listed in a Rothermel paper... Maybe I should put it back?

Reply:
and yes I agree I can use that method for the equation.

Reply:
It does make sense, especially if it was in a Rothermel paper too. But I wonder if we want to be making changes like this in a PR that's otherwise purely refactoring.

Reply:
Okay removing it!